roles/upgrade/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+- name: "Stop the OpenLDAP Server"
+  ansible.builtin.systemd:
+    name: slapd
+    daemon_reload: true
+    enabled: true
+    state: stopped
+
+- name: Backup LDAP DB
+  ansible.builtin.command: slapcat -l /tmp/ldap-backup.ldif
+  become: yes
+  args:
+    creates: /tmp/ldap-backup.ldif
+
+- name: Clear OpenLDAP DB directory
```

Comment:
```yaml
- name: Ensure backup exists before deleting DB
  ansible.builtin.stat:
    path: /tmp/ldap-backup.ldif
  register: ldap_backup
  
  - name: Fail if backup not found
  ansible.builtin.fail:
    msg: "Backup file not found. Aborting to prevent data loss."
  when: not ldap_backup.stat.exists
 ```